### PR TITLE
Updating Views

### PR DESCRIPTION
**Pull request by camidemery**
Increasing efficiency

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -2,16 +2,16 @@ class ItemsController < ApplicationController
     def index
         @items=Item.all
     end
-    
+
     def show
         @item=Item.find(params[:id])
     end
```
Comment (leila-alderman) [inline app/controllers/items_controller.rb:8]: [Style] Ruby code should be [indented 2 spaces, not 4](https://github.com/rubocop-hq/ruby-style-guide#spaces-indentation).

You can change this setting in your IDE to make it the default for Ruby files.

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -24,5 +24,5 @@ def create
         @item.save
         redirect_to items_path
     end
```
Comment (leila-alderman) [inline app/controllers/items_controller.rb:26]: Please update all of the controllers to use strong parameters. If you need some guidance on this, let me know. Thanks!

File: app/controllers/procedures_controller.rb [hunk 3]
```diff
@@ -31,12 +31,12 @@ def edit
 
     #update exisiting database entry -- only available to admin? -------
     def update
-        procedure = Procedure.new
+        procedure = Procedure.find
```
Comment (leila-alderman) [inline app/controllers/procedures_controller.rb:34]: The `.find` method requires an argument.

File: app/views/facilities/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<%=form_for @facility do |f|%>
+    <div class="p-2">
+        <%=f.text_field  :name ,placeholder:"Name of Facility"%>
+    </div>
+    <div class="p-2">
+        <%=f.text_field :city ,placeholder:"City"%>
+    </div>
```
Comment (leila-alderman) [inline app/views/facilities/_form.html.erb:7]: To add an HTML class to a Rails form, you can do it in a more Rails-y way with cleaner syntax (allowing you to remove all of the `div`s):

`<%= f.text_field  :name ,placeholder: "Name of Facility", class: 'p-2' %>`

File: app/views/facilities/edit.html.erb [hunk 5]
```diff
@@ -1,25 +1,7 @@
...
-    </div>
+<div class="container border border-secondary rounded-lg mt-3">
+  <h1 class="text-center">
+      Edit Facility Information
+  </h1>
+</div>
 
-<%end%>
\ No newline at end of file
+<%= render 'form' %>
```
Comment (leila-alderman) [inline app/views/facilities/edit.html.erb:7]: Nice change!

File: app/views/facilities/index.html.erb [hunk 6]
```diff
@@ -1,26 +1,25 @@
...
-  </a>
-  <div class='list-group-item-small align-middle p-3 col-2'>
-    <%=link_to "Edit",edit_facility_path(f.id)%> | <%= link_to "Delete",facility_path(f.id), method: :delete, data: { confirm: 'Are you sure?'} %>
-  </div>
-</ul>
-  <%end%>
+  <ul class="list-group row p-2">
+    <% @facilities.each do |f|%>
+      <a href="/facilities/<%=f.id%>" class="list-group-item list-group-item-action col-4">
+        <h4 class=''><%=f.name%></h4>
```
Comment (leila-alderman) [inline app/views/facilities/index.html.erb:15]: If the class here is empty, you should be able to remove it altogether
```suggestion
        <h4><%=f.name%></h4>
```

File: app/views/items/_form.html.erb [hunk 7]
```diff
@@ -1,18 +1,21 @@
...
-          <% end %>
-        </option>
-      </select>
-      <%= link_to  "Add Facility",new_facility_path %>
-    </div>
-<% end %> 
+  <div class="p-2"><%= f.text_field :name, placeholder: 'Name' %></div>
+  <div class="p-2"><%= f.date_field :date, placeholder: 'Date' %></div>
+  <div class="p-2"><%= f.text_field :description, placeholder: 'Description' %></div>
+  <div class="p-2"><%= f.number_field :price, placeholder: 'Price', min: 0, step: 0.01 %></div>
```
Comment (leila-alderman) [inline app/views/items/_form.html.erb:5]: As mentioned above, you can make these form statements more succinct by moving the `class` into the form helper.

File: app/views/items/_form.html.erb [hunk 8]
```diff
@@ -1,18 +1,21 @@
...
+  <div class="p-2">
+     <select name='facility_id'>
+       <option>
+         <% @facilities.each do |f| %>
+           <option value="<%= f.id %>", placeholder:"Facility"><%= f.name %></option>
+         <% end %>
+       </option>
+     </select>
```
Comment (leila-alderman) [inline app/views/items/_form.html.erb:13]: Check out this Stack Overflow post on [how to create a dropdown in Rails](https://stackoverflow.com/questions/14113057/how-to-have-a-drop-down-select-field-in-a-rails-form).

File: app/views/items/new.html.erb [hunk 9]
```diff
@@ -1,3 +1,9 @@
...
\ No newline at end of file
+<div class="container border border-secondary rounded-lg mt-3">
+  <h1 class="text-center">
+      New Item
+  </h1>
+</div>
+
+<container>
+  <%= render 'form' %>
+</container>
```
Comment (leila-alderman) [inline app/views/items/new.html.erb:9]: What do the `<container>` tags here do? (I'm wondering if they can be deleted or if they're needed for something else that's happening.)

File: app/views/procedures/_form.html.erb [hunk 10]
```diff
@@ -1,16 +1,22 @@
...
+        <option value="<%= f.id %>"><%= f.name %></option>
+     <% end %>
+   </option>
+  </select>
  </div>
- <%= p.submit "Submit"%>
- <%end%>
+<div class="p-2">
+  <%= p.submit 'Submit', class:'btn btn-outline-primary'%>
+</div>
```
Comment (leila-alderman) [inline app/views/procedures/_form.html.erb:17]: You should be able to add the `class='p-2'` to the list of classes already on the submit button to simplify the code.

File: app/views/procedures/edit.html.erb [hunk 11]
```diff
@@ -1,2 +1,9 @@
...
-<%= render 'form' %>
+<div class="container border border-secondary rounded-lg mt-3">
+  <h1 class="text-center">
+      Edit Procedure
+  </h1>
+</div>
+
+<container>
+  <%= render 'form' %>
+</container>
```
Comment (leila-alderman) [inline app/views/procedures/edit.html.erb:9]: Again, are these `<container>` tags needed?

File: app/views/procedures/index.html.erb [hunk 12]
```diff
@@ -1,21 +1,37 @@
...
+    <div class="container">
+      <div class="row">
+      <%= @procedures.each do |p|%>
+      <div class="col-md-3 ">
+        <div class="card mb-3">
+          <div class="card-header" align="center"><%=p.name%></div>
+          <div class="card-body" align='left'>
+            <%=p.description%>
+            <div>
+            </div>
```
Comment (leila-alderman) [inline app/views/procedures/index.html.erb:23]: Can this empty `<div>` be deleted?

File: app/views/procedures/new.html.erb [hunk 13]
```diff
@@ -1,2 +1,9 @@
...
-<%= render 'form' %>
+<div class="container border border-secondary rounded-lg mt-3">
+  <h1 class="text-center">
+      New Procedure
+  </h1>
+</div>
+
+<container>
+  <%= render 'form' %>
+</container>
```
Comment (leila-alderman) [inline app/views/procedures/new.html.erb:9]: Again, are these `<container>` tags needed?

**Review by leila-alderman — CHANGES_REQUESTED** [review]
The only two major things that need to be changed are updating to strong parameters and providing an argument for `Procedure.find` in the Procedures controller.

My other comments are more minor notes related to code style and simplifying. 

As always, please let me know if you have any questions or want to discuss further!

**Comment by camidemery** [comment]
Going to re-update views after models are redone

**Pull request closed without merging**